src/components/Checkbox/Checkbox.tsx

```diff
@@ -43,37 +51,25 @@ export type CheckboxProps = Omit<CheckboxInputProps, 'id'> & {
  *
  * NOTE: Requires either a visible label or `aria-label` prop.
  */
-export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
-  (props, ref) => {
-    // All remaining props are passed to the `input` element
-    const { className, id, label, size = 'lg', disabled, ...other } = props;
+export const Checkbox: CheckboxType = React.forwardRef((props, ref) => {
```

Comment:
nit: usually we have an import of `forwardRef` directly, so we could tidy up the imports here

Reply:
Dried!